src/measure_theory/bochner_integration.lean

```diff
@@ -16,32 +45,537 @@ set_option class.instance_max_depth 100
+    { rw [a0, hg0, zero_mul], exact with_top.zero_lt_top },
+    apply mul_lt_top (hgt a) (hf' _ a0) },
+  { simp [hg0] }
+end
+
+/-- `simple_func.bintegral` and `lintegral : (α → ennreal) → ennreal` are the same when the
+    integrand has type `α →ₛ ennreal`. But since `ennreal` is not a `normed_space`, we need some
+    form of coercion.
+    See `bintegral_eq_lintegral'` for a simpler version. -/
+lemma bintegral_eq_lintegral (f : α →ₛ β) (g : β → ennreal) (hf : integrable f) (hg0 : g 0 = 0)
+  (hgt : ∀b, g b < ⊤):
+  (f.map (ennreal.to_real ∘ g)).bintegral = ennreal.to_real (∫⁻ a, g (f a)) :=
+by { rw [bintegral_eq_integral hf hg0 hgt, ← lintegral_eq_integral], refl }
+
+variables [normed_space ℝ β]
+
+lemma bintegral_congr {f g : α →ₛ β} (hf : integrable f) (hg : integrable g) (h : ∀ₘ a, f a = g a):
+  bintegral f = bintegral g :=
+show ((pair f g).map prod.fst).bintegral = ((pair f g).map prod.snd).bintegral, from
+begin
+  have inte := integrable_pair hf hg,
+  rw [map_bintegral (pair f g) _ inte prod.fst_zero, map_bintegral (pair f g) _ inte prod.snd_zero],
+  refine finset.sum_congr rfl (assume p hp, _),
+  rcases mem_range.1 hp with ⟨a, rfl⟩,
+  by_cases eq : f a = g a,
+  { dsimp only [pair_apply], rw eq },
+  { have : volume ((pair f g) ⁻¹' {(f a, g a)}) = 0,
+    { refine volume_mono_null (assume a' ha', _) h,
+      simp only [set.mem_preimage, mem_singleton_iff, pair_apply, prod.mk.inj_iff] at ha',
+      show f a' ≠ g a',
+      rwa [ha'.1, ha'.2] },
+    simp only [this, pair_apply, zero_smul, ennreal.zero_to_real] },
+end
+
+/-- `simple_func.bintegral` and `simple_func.integral` agrees when the integrand has type
+    `α →ₛ ennreal`. But since `ennreal` is not a `normed_space`, we need some form of coercion. -/
+lemma bintegral_eq_integral' {f : α →ₛ ℝ} (hf : integrable f) (h_pos : ∀ₘ a, 0 ≤ f a) :
+  f.bintegral = ennreal.to_real (f.map ennreal.of_real).integral :=
+begin
+  have : ∀ₘ a, f a = (f.map (ennreal.to_real ∘ ennreal.of_real)) a,
+  { filter_upwards [h_pos],
+    assume a,
+    simp only [mem_set_of_eq, map_apply, function.comp_apply],
+    assume h,
+    exact (ennreal.to_real_of_real h).symm },
+  rw ← bintegral_eq_integral hf,
+  { refine bintegral_congr hf _ this, exact integrable_of_ae_eq hf this },
+  { exact ennreal.of_real_zero },
+  { assume b, rw ennreal.lt_top_iff_ne_top, exact ennreal.of_real_ne_top  }
+end
+
+/-- `simple_func.bintegral` and `lintegral : (α → ennreal) → ennreal` agrees when the integrand has
+    type `α →ₛ ennreal`. But since `ennreal` is not a `normed_space`, we need some form of coercion. -/
+lemma bintegral_eq_lintegral' {f : α →ₛ ℝ} (hf : integrable f) (h_pos : ∀ₘ a, 0 ≤ f a) :
+  f.bintegral = ennreal.to_real (∫⁻ a, (f.map ennreal.of_real a)) :=
+by rw [bintegral_eq_integral' hf h_pos, ← lintegral_eq_integral]
+
+lemma bintegral_add {f g : α →ₛ β} (hf : integrable f) (hg : integrable g) :
+  bintegral (f + g) = bintegral f + bintegral g :=
+calc bintegral (f + g) = _ :
```

Comment:
Could you write the content of the underscore on the right? I know Lean does not need it, but it would help the human readers! Same thing a few lines below.